server_test.go

```diff
@@ -0,0 +1,77 @@
+package osquery
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+	"testing"
+
+	"github.com/kolide/osquery-go/gen/osquery"
+	"github.com/kolide/osquery-go/plugin/logger"
+	"github.com/stretchr/testify/assert"
+)
+
+type mockExtensionClient struct {
+	mut    sync.Mutex
```

Comment:
The only reason this lock is there is to prevent the race detector from complaining when you run tests. This is because of the mock.called member, which I think I need to have to verify that RegisterExtension has in fact been invoked.